homeassistant/components/airthings_ble/config_flow.py

```diff
@@ -0,0 +1,144 @@
+"""Config flow for Airthings BlE integration."""
+
+from __future__ import annotations
+
+import dataclasses
+import logging
+from typing import Any
+
+from airthings_ble import AirthingsBluetoothDeviceData, AirthingsDevice
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfo,
+    async_discovered_service_info,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class Discovery:
+    """A discovered bluetooth device."""
+
+    name: str
+    discovery_info: BluetoothServiceInfo
+    device: AirthingsDevice
+
+
+def get_name(device: AirthingsDevice) -> str:
+    """Generate name with identifier for device."""
+    return f"{device.name} {device.identifier}"
+
+
+class AirthingsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Airthings BLE."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovered_device: Discovery | None = None
+        self._discovered_devices: dict[str, Discovery] = {}
+
+    async def _get_device_data(self, discovery_info: BluetoothServiceInfo):
+        ble_device = bluetooth.async_ble_device_from_address(
+            self.hass, discovery_info.address.upper()
+        )
+        if ble_device is None:
+            raise UpdateFailed(f"No device for {discovery_info.address} found")
+
+        airthings = AirthingsBluetoothDeviceData(_LOGGER)
+
+        try:
+            data = await airthings.update_device(ble_device)
+        except Exception as err:
+            raise UpdateFailed(f"Unable to fetch data: {err}") from err
+
+        return data
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug("Discovered BT device: %s", discovery_info)
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+
+        if 820 not in discovery_info.manufacturer_data:
+            return self.async_abort(reason="not_supported")
```

Comment:
```suggestion
```

This code is unreachable since you'll never get a discovery without it based on your `manifest.json`